File: pinyougou/pinyougou-manage-web/src/main/webapp/js/service/itemCatService.js
// 服务层
app.service("itemCatService", function ($http) {
    //获得全部itemCat
    this.findAll = function () {
        return $http.post("../itemCat/findAll.do");
    };
    //通过父类id查询下级
    this.findByParentId = function (parentId) {
        return $http.get("../itemCat/findByParentId.do?parentId="+parentId);
    };


});

File: pinyougou/pinyougou-sellergoods/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbItemCat;
import com.pinyougou.service.BaseService;

import java.util.List;

public interface ItemCatService extends BaseService<TbItemCat> {


    List<TbItemCat> findByParentId(Long parentId);
}